### ISSUE TITLE
Enhancement: Roborock Video-Stream without an Video-Call

### ISSUE THREAD

**Issue by superpower10** [AlexxIT/go2rtc#475]
go2rtc establishes a video call to the Roborock S7. This stops its from working and continues after ending the „call“.

Would like to establish a video streaming to the Roborock S7 like the App does without stopping it.

**Comment by AlexxIT** [comment]
This depends on how you get stream. You should get stream without audio. For example RTSP with only video filter or webrtc.html with only video media

**Comment by superpower10** [comment]
No. That doesn‘t work. It still starts a video call and stops working.

type: custom:webrtc-camera
url: roborock://…
muted: true

**Comment by mang1985** [comment]
> @AlexxIT Sadly, I couldn’t get it to work that way in Home Assistant. It still starts a video call and stops working.
> 
> How do I have to configure it correctly?
> 
> Thank you


If you are using the 3.2.1 version of the WebRTC Camera integration in Homeassistant and encounter issues, try reinstalling version 3.2.0. This is how I resolved it. @superpower10

**Comment by mang1985** [comment]
> > @AlexxIT Sadly, I couldn’t get it to work that way in Home Assistant. It still starts a video call and stops working.
> > How do I have to configure it correctly?
> > Thank you
> 
> If you are using the 3.2.1 version of the WebRTC Camera integration in Homeassistant and encounter issues, try reinstalling version 3.2.0. This is how I resolved it. @superpower10

edit: Today found that the problem might be caused by the go2rtc addon version 1.6.2. If you have a backup, you can try downgrading it. Using WebRTC Camera integration 3.2.0 and go2rtc addon 1.5.0 could be a viable solution.

**Comment by madasus** [comment]
I have the same issue and I went all the way back to go2rtc 1.3 and still have the issue.

I'm not triggering the feed from HA i'm doing it directly from the go2rtc interface on port 1984 so I would think this eliminates the problem being in the WebRTC integration.

**Comment by rodolfomachado** [comment]
So the solution is downgrading?

**Comment by xJylin** [comment]
Having the same issue with my Roborock S7 MaxV.
go2rc version 1.9.3

Anyone found a solution for this or can give a example configuration that works?

**Comment by Sonic-Hedgehog** [comment]
For those who haven’t found the solution yet: Simply add `?video` to the URL of the RTSP stream. This will open the **Remote View** instead of starting the video call.  

⚠️ Make sure that the security PIN for the camera is either disabled in the Roborock app or correctly included in the URL. You can find the security PIN in the Roborock app under:  
**Settings > Remote Viewing > Security Settings**  

### Example URLs:  
- **Without PIN:**  
  `rtsp://localhost:8554/camera.roborock?video`  

- **With PIN:**  
  `rtsp://localhost:8554/camera.roborock?video&pin=1234`  

### Note on the PIN structure:  
The PIN corresponds to the pattern code of the unlock gesture and is based on the following layout:  
```
1 2 3  
4 5 6  
7 8 9  
```  

Example: If your pattern is a line from the top left (1) through the middle (5) to the bottom right (9), the PIN will be `159`.  

### Troubleshooting:  
If an error occurs, please ensure that the camera is functioning correctly in the Roborock app. For example, check if the camera displays without issues in the app and ensure that Remote Viewing is enabled.

**Comment by piotr3kg** [comment]
I'm trying to figure it out. First I've installed only go2rtc addon and video previev worked well with only video view on my Roborock S7MaxV. I was able to watch the stream during cleaning in go2rtc tab in HA. Then I wanted to add this video preview to dashboard under my vaccum card. So I've installed the WebRTC Camera addon to get the Lovelace custom card. Now I'm having the issue mencioned in this thread. Mean that I'm getting only video call instead video preview. I'm unable to go back to situation from the start even after removing WebRTC camera addon and reinstaling go2rtc alone. Now when I turn on the stream the phone call starts instead video preview. I've tried solution from the post above but no success. Any other sugetions? Sorry for my poor english.

**Comment by rodolfomachado** [comment]
Never worked here. Even with the last suggestions.
Here it pauses the cleaning just after starting streaming the video

**Comment by piotr3kg** [comment]
I would like to share something that i've managed to achieve after few attempts. I think it is worknig in my case as I wanted to. I've added a WebRTC lovelace custom card and this is the code I've added.

type: custom:webrtc-camera
url: >-
  roborock://mqtt-eu.roborock.com:XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXpin=XXXXX#media=video
media: video

After HA restart now when my vaccum is operating I can see live picture on my dashboard and no more phone call instead. Mayby this can be useful for someone.

**Comment by rodolfomachado** [comment]
Wow! Now it is working!

Tks a lot!


Em ter., 25 de mar. de 2025, 15:35, piotr3kg ***@***.***>
escreveu:
